conda/core/path_actions.py

```diff
@@ -300,29 +301,11 @@ def verify(self):
             )
 
         elif source_path_data.path_type == PathType.hardlink:
-            try:
-                reported_sha256 = source_path_data.sha256
-            except AttributeError:
-                reported_sha256 = None
-            source_sha256 = compute_sha256sum(self.source_full_path)
-            if reported_sha256 and reported_sha256 != source_sha256:
-                return SafetyError(dals("""
-                The package for %s located at %s
-                appears to be corrupted. The path '%s'
-                has a sha256 mismatch.
-                  reported sha256: %s
-                  actual sha256: %s
-                """ % (self.package_info.repodata_record.name,
-                       self.package_info.extracted_package_dir,
-                       self.source_short_path,
-                       reported_sha256,
-                       source_sha256,
-                       )))
-
             try:
                 reported_size_in_bytes = source_path_data.size_in_bytes
             except AttributeError:
                 reported_size_in_bytes = None
+                source_size_in_bytes = 0
```

Comment:
I think this needs to one less indentation level.  As it is source_size_in_bytes may be undefined at line 330 if repored_size_in_bytes is 0.